content/advent-2018/go-in-the-browser.md

```diff
@@ -0,0 +1,175 @@
++++
+author = ["Agniva De Sarker"]
+title = "Go and WebAssembly: running Go programs in your browser"
+linktitle = "Go in the browser"
+date = 2018-12-07T00:00:00Z
+series = ["Advent 2018"]
++++
+
+For a long time, Javascript was the lingua franca amongst web developers. If you wanted to write a stable, mature webapp, writing in javascript was pretty much the only way to go.
+
+WebAssembly is going to that change soon. Using WebAssembly (also called wasm), one can write webapps in _any_ language now. In this post, we will see how to write Go programs and run them in the browser using wasm.
+
+## But first, what is WebAssembly
+
+The [webassembly.org](https://webassembly.org/) site defines it as "WebAssembly is a binary instruction format for a stack-based virtual machine". That is a great definition, but let us break it down to something we can easily understand.
+
+Essentially, wasm is a binary format; just like ELF, Mach, and PE. The only difference is that it is for a virtual compilation target, not an actual physical machine. Why virtual? because unlike C/C++ binaries, wasm binaries are not targeted for a specific platform. So you can use the same binary in Linux, Windows and Mac without changing anything. As a result, we need another "agent" which translates the wasm instructions inside the binary into platform specific instructions and actually run them. This "agent" is actually the browser. But, in theory it can just as well be anything else.
```

Comment:
The semicolon is technically correct here, because the sentence "Essentially, wasm is a binary format" stands on its own. Also, adding a comma after "Because" is not necessary I believe.